src/modules/aptextrapackages/SetSourcesJob.cpp

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+ * Copyright (c) 2016-2017, Kyle Robbertze (AIMS, South Africa)
+ *
+ * This project is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * It is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this project. If not, see <http://www.gnu.org/licenses/>.
+ *
+ *    Description:  Module for installing select packages in the installed
+ *                  system
+ *
+ *        Created:  09/12/2016 09:35:39
+ *       Compiler:  g++
+ *
+ *         Author:  Kyle Robbertze (kr), kyle@aims.ac.za
+ *******************************************************************************/
+
+#include "SetSourcesJob.h"
+
+#include "GlobalStorage.h"
+#include "JobQueue.h"
+#include "utils/CalamaresUtilsSystem.h"
+
+#include <QFile>
+#include <QTextStream>
+
+SetSourcesJob::SetSourcesJob( const QString& source ) :
+    Calamares::Job(),
+    m_source( source )
+{ }
+
+QString SetSourcesJob::prettyName() const
+{
+    return tr( "Add source %1" ).arg( m_source );
+}
+
+QString SetSourcesJob::prettyDescription() const
+{
+    return tr( "Add <strong>%1</strong> to apt sources list" ).arg( m_source );
+}
+
+QString SetSourcesJob::prettyStatusMessage() const
+{
+    return tr( "Adding %1 to sources list" ).arg( m_source );
+}
+
+Calamares::JobResult SetSourcesJob::exec()
```

Comment:
Just curious, why does Calamares need to add stuff to sources.list? Can't the source list ship already configured on the ISO?